### DISCUSSION TITLE
LXQt applications in wayland has empty app_id

### DISCUSSION THREAD

**Discussion by romildo** [lxqt/lxqt#2559] [General]
In NixOS, LXQt applications running on wayland (verified with labwc, hyprland) have empty app_id.

See https://github.com/NixOS/nixpkgs/issues/306366

This is the labwc window switcher:
![image](https://github.com/lxqt/lxqt/assets/1217934/07bf2a1f-5e3b-48f9-96f0-0291ddef5075)
Note that the app_id field (penultimate field) is empty for Qt applications (lxqt-archiver, lximage-qt, pcmanfm-qt, qalculateqt).

Has anyone seen this issue? Any clues?

**Comment by tsujan** [comment]
> Has anyone seen this issue?

Those that I have here have app-id's, under LabWC:

![app-id](https://github.com/lxqt/lxqt/assets/981076/c8422c5e-1bb5-4752-816e-5689e7586848)

**Reply by Schweber** [reply]
Are you using Nixos?

**Reply by stefonarch** [reply]
This in fine on arch and derivatives and under debian bookworm too, so it's a Nixos issue.

```
$ lswt
app-id:         title:
qterminal       "stef@archlinux: ~/git/lxqt/screengrab"
lxqt-archiver   "Gestore di archivi"
pcmanfm-qt      screengrab
qps             stef@archlinux
lximage-qt      "Visualizzatore immagini"
```
